### PR TITLE
WebApi : Added BaseId to Item

### PR DESCRIPTION
**Pull request by namidaka**

### CODE REVIEW

File: src/Persistence/Migrations/20230604201330_BaseId.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
```
Comment (verdie-g) [inline src/Persistence/Migrations/20230604201330_BaseId.cs:4]: Can be removed

File: src/Persistence/Migrations/20230604201330_BaseId.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+namespace Crpg.Persistence.Migrations;
+
+/// <inheritdoc />
+public partial class BaseId : Migration
```
Comment (verdie-g) [inline src/Persistence/Migrations/20230604201330_BaseId.cs:8]: Could you give a more explicit name to the migration such as "AddBaseItemToItem" please

File: test/Application.UTest/Games/GetGameUserCommandTest.cs [hunk 3]
```diff
@@ -413,7 +413,7 @@ public async Task CheckDefaultItemsExistAndAreNotTooExpensive()
                 price += item.Price;
             }
 
-            Assert.That(price, Is.LessThan(2900));
+            Assert.That(price, Is.LessThan(3500));
```
Comment (verdie-g) [inline test/Application.UTest/Games/GetGameUserCommandTest.cs:416]: What happened there 🤔

File: src/Domain/Entities/Items/UserItem.cs [hunk 4]
```diff
@@ -11,6 +11,7 @@ public class UserItem : AuditableEntity
     public int Id { get; set; }
     public int UserId { get; set; }
     public string ItemId { get; set; } = string.Empty;
+    public string BaseItemId { get; set; } = string.Empty;
```
Comment (verdie-g) [inline src/Domain/Entities/Items/UserItem.cs:14]: This column is redundant with `.Item.BaseItemId`

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: test/Application.UTest/Games/GetGameUserCommandTest.cs (hunk 3 above)
Comment (namidaka) [reply test/Application.UTest/Games/GetGameUserCommandTest.cs:416]: New Items.json with baseId  is in this pullrequest. We will take care of lowering that , but for the moment i did not feel like it was in the scope of this pr

**Review by verdie-g — CHANGES_REQUESTED** [review]
Could you also update the item ids in https://github.com/verdie-g/crpg/blob/master/src/Module.Server/ModuleData/characters.xml

File: src/Persistence/Migrations/20230604205100_AddedBaseIdToItem.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    protected override void Up(MigrationBuilder migrationBuilder)
+    {
+        migrationBuilder.DropForeignKey(
+            name: "fk_user_items_items_base_item_id",
+            table: "user_items");
+
+        migrationBuilder.RenameColumn(
+            name: "base_item_id",
+            table: "user_items",
+            newName: "item_id");
```
[the remark this answers is not in the document]
Comment (namidaka) [reply src/Persistence/Migrations/20230604205100_AddedBaseIdToItem.cs:18]: It was not like for your microsoft migration. I don't know why.

[the remark this answers is not in the document]
Comment (namidaka) [reply src/Persistence/Migrations/20230604205100_AddedBaseIdToItem.cs:18]: i proposed a separate pr for fixing this before : https://github.com/verdie-g/crpg/pull/1008

**Comment by namidaka** [comment]
> Could you also update the item ids in https://github.com/verdie-g/crpg/blob/master/src/Module.Server/ModuleData/characters.xml

This one was a true pain in the ass. It's done

File: src/Persistence/Migrations/20230604205100_AddedBaseIdToItem.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (verdie-g) [reply src/Persistence/Migrations/20230604205100_AddedBaseIdToItem.cs:18]: Was the branch based on an up-to-date master when you created the migration? 🤔

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Persistence/Migrations/20230604205100_AddedBaseIdToItem.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (namidaka) [reply src/Persistence/Migrations/20230604205100_AddedBaseIdToItem.cs]: I just redid the migration after pulling master and rebasing on master. It did not change anything. As i said , the file dbcontextsnapshot was no changed in the last commit to master. Please look at https://github.com/verdie-g/crpg/pull/1008

File: src/Persistence/Migrations/CrpgDbContextModelSnapshot.cs
Comment (verdie-g) [inline src/Persistence/Migrations/CrpgDbContextModelSnapshot.cs:1]: Can you discard the changes here

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Persistence/Migrations/CrpgDbContextModelSnapshot.cs
Comment (namidaka) [reply src/Persistence/Migrations/CrpgDbContextModelSnapshot.cs:1]: done

File: src/Application/Items/Models/ItemCreation.cs [hunk 6]
```diff
@@ -6,6 +6,8 @@ namespace Crpg.Application.Items.Models;
 public record ItemCreation
 {
     public string Id { get; init; } = string.Empty;
+    public string BaseId { get; init; } = string.Empty;
+
```
Comment (verdie-g) [inline src/Application/Items/Models/ItemCreation.cs:10]: Also if you could remove this line

File: src/Domain/Entities/Items/Item.cs [hunk 7]
```diff
@@ -11,6 +11,8 @@ public class Item : AuditableEntity, ICloneable
     /// Same id as the Mount & Blade item.
     /// </summary>
     public string Id { get; set; } = string.Empty;
+    public string BaseId { get; set; } = string.Empty;
+
```
Comment (verdie-g) [inline src/Domain/Entities/Items/Item.cs:15]: And this line

File: src/Persistence/Migrations/CrpgDbContextModelSnapshot.cs
Comment (verdie-g) [inline src/Persistence/Migrations/CrpgDbContextModelSnapshot.cs:1]: Still there

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Persistence/Migrations/CrpgDbContextModelSnapshot.cs
Comment (namidaka) [reply src/Persistence/Migrations/CrpgDbContextModelSnapshot.cs:1]: Completely done this time

File: src/Domain/Entities/Items/Item.cs (hunk 7 above)
Comment (namidaka) [reply src/Domain/Entities/Items/Item.cs]: done

File: src/Application/Items/Models/ItemCreation.cs (hunk 6 above)
Comment (namidaka) [reply src/Application/Items/Models/ItemCreation.cs]: done

File: src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace Crpg.Persistence.Migrations
+{
```
Comment (verdie-g) [inline src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: Use file-scoped namespace

File: src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "base_id",
+                table: "items",
+                type: "text",
+                nullable: false,
+                defaultValue: "");
```
Comment (verdie-g) [inline src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: Warning to fix

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs (hunk 8 above)
Comment (namidaka) [reply src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: and i pushed ! 
image

Comment (namidaka) [reply src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: There is something i'm clearly not understanding there. I removed the nullabe disable and scoped the namespaces then pushed , yet it still shows up like i didn't?!

File: src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs [hunk 10]
```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
```
[the remark this answers is not in the document]
Comment (namidaka) [reply src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: Wait? What did you comment? It's like already outdated . when the last change i did was 40 minute before your comment

File: src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs (hunk 8 above)
Comment (namidaka) [reply src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: Actually this is not on my side. My last commit was 40 minute before your comment yet your comment was already outdated when you made it

File: src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (namidaka) [reply src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: image

[the remark this answers is not in the document]
Comment (namidaka) [reply src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: Oh i get it . You commit the specific Migration Commit . i did two commit right next to each other :D 
image

File: src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs (hunk 8 above)
Comment (namidaka) [reply src/Persistence/Migrations/20230606122412_AddBaseIdToItems.cs]: I get it now you commented a specific commit , but i did two commit one after the other : 
image

**Review by verdie-g — APPROVED** [review]

**Pull request merged**